composer/loggers/slack_logger.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Callable, Dict, Optional
+
+from slack_sdk import WebhookClient
+
+from composer.loggers.logger_destination import LoggerDestination
+
+
+class SlackLogger(LoggerDestination):
+    """Log metrics to slack, using `Slack Webhook URL <https://api.slack.com/messaging/webhooks>`
+    
+    .. note::
+        See example instantiating of SlackLogger in `examples/slack_log_metrics.py`
+        Webhook URL will be passed as an environment variable.
+        Formatter function can be defined at callsite or default formatter function will be applied if not provided.
+         
+    Args:
+        webhook_url (str): Slack Webhook URL
+        log_metrics_formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack
+    """
+
+    def __init__(
+        self,
+        webhook_url: str,
+        log_metrics_formatter_func: Optional[Callable] = None,
+    ) -> None:
+        self.webhook_url = webhook_url
```

Comment:
What do you think of webhook url being optional if the user sets the SLACK_WEBHOOK_URL env variable. Then it would be a lot easier for someone to write a script because they wouldn't have to go look for it every time or save it to their script